external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaSource.scala

```diff
@@ -149,6 +160,14 @@ private[kafka010] case class KafkaSource(
         Some(sortedExecutors(positiveMod(tp.hashCode, numExecutors)))
       } else None
       KafkaSourceRDDOffsetRange(tp, fromOffset, untilOffset, preferredLoc)
+    }.filter { range =>
+      if (range.untilOffset < range.fromOffset) {
+        logWarning(s"Partition ${range.topicPartition} was deleted and then added, " +
```

Comment:
I'd change this to error level and specify what the offsets in question were.  I also wouldn't assume that a partition was deleted and then added, this could be caused by other things.